src/dodal/devices/aperturescatterguard.py

```diff
@@ -14,85 +17,134 @@ class InvalidApertureMove(Exception):
+                    params["sg_y_MEDIUM_APERTURE"],
+                ),
             ),
-            SMALL=(
-                params["miniap_x_SMALL_APERTURE"],
-                params["miniap_y_SMALL_APERTURE"],
-                params["miniap_z_SMALL_APERTURE"],
-                params["sg_x_SMALL_APERTURE"],
-                params["sg_y_SMALL_APERTURE"],
+            SMALL=SingleAperturePosition(
+                "Small",
+                20,
+                ApertureFiveDimensionalLocation(
+                    params["miniap_x_SMALL_APERTURE"],
+                    params["miniap_y_SMALL_APERTURE"],
+                    params["miniap_z_SMALL_APERTURE"],
+                    params["sg_x_SMALL_APERTURE"],
+                    params["sg_y_SMALL_APERTURE"],
+                ),
             ),
-            ROBOT_LOAD=(
-                params["miniap_x_ROBOT_LOAD"],
-                params["miniap_y_ROBOT_LOAD"],
-                params["miniap_z_ROBOT_LOAD"],
-                params["sg_x_ROBOT_LOAD"],
-                params["sg_y_ROBOT_LOAD"],
+            ROBOT_LOAD=SingleAperturePosition(
+                "Robot load",
+                None,
+                ApertureFiveDimensionalLocation(
+                    params["miniap_x_ROBOT_LOAD"],
+                    params["miniap_y_ROBOT_LOAD"],
+                    params["miniap_z_ROBOT_LOAD"],
+                    params["sg_x_ROBOT_LOAD"],
+                    params["sg_y_ROBOT_LOAD"],
+                ),
             ),
         )
 
-    def position_valid(self, pos: Tuple[float, float, float, float, float]):
+    def get_new_position(
+        self, pos: ApertureFiveDimensionalLocation
+    ) -> SingleAperturePosition:
         """
         Check if argument 'pos' is a valid position in this AperturePositions object.
         """
-        if pos not in [self.LARGE, self.MEDIUM, self.SMALL, self.ROBOT_LOAD]:
-            return False
-        return True
+        options: List[SingleAperturePosition] = [
+            self.LARGE,
+            self.MEDIUM,
+            self.SMALL,
+            self.ROBOT_LOAD,
+        ]
+        pos_list = list(pos)
+        for obj in options:
+            local_position = list(obj.location)
+            if np.allclose(local_position, pos_list, atol=self.TOLERANCE_MM):
+                return obj
+        return None
 
 
 class ApertureScatterguard(InfoLoggingDevice):
     aperture = Cpt(Aperture, "-MO-MAPT-01:")
     scatterguard = Cpt(Scatterguard, "-MO-SCAT-01:")
     aperture_positions: Optional[AperturePositions] = None
+    selected_aperture = Cpt(Signal)
     APERTURE_Z_TOLERANCE = 3  # Number of MRES steps
 
     def load_aperture_positions(self, positions: AperturePositions):
         LOGGER.info(f"{self.name} loaded in {positions}")
         self.aperture_positions = positions
 
-    def set(self, pos: Tuple[float, float, float, float, float]) -> AndStatus:
+    def set(self, pos: ApertureFiveDimensionalLocation) -> AndStatus:
```

Comment:
Should: Why not just set a `SingleAperturePosition` here? It simplifies things as we don't have to work out which position it is based on the values sent and we can just check it is equal to one of the objects defined in `AperturePositions`

Reply:
hmm that sounds good. I was more hesitant to change this external API of the class as opposed to refactoring it internally

Reply:
> Override the read on the selected_aperture such that when you read from it it reads all the motor positions and works out which is selected from that

I am not sure how to do that

Reply:
will be extracted into a new ticket